Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -42,7 +42,7 @@ void doDepositionShapeN(const amrex::Real * const xp,
     const amrex::Real dts2dz = 0.5*dt*dzi;
 #if (AMREX_SPACEDIM == 2)
     const amrex::Real invvol = dxi*dzi;
-#else // (AMREX_SPACEDIM == 3)
+#elif (defined WARPX_DIM_3D)
     const amrex::Real dyi = 1.0/dx[1];
```

Comment:
My understanding is that `doDepositionShapeN` will not work in `RZ` mode, is that correct?
Could you add the `RZ` implementation as part of this PR?

Reply:
I can do this also, though is the direct deposition needed in RZ? My understanding is that direct deposition is only needed for the spectral solver. For spectral RZ, the deposition will be enough different (with the complex modes) that separate code will likely be needed.

Reply:
Hm, I think it is worth adding the direct deposition in RZ, since it does not seem to be a big change in the code.
My understanding is that the only changes would be: 
- Redefining `xmid` (in the same way as `x_old` and `x_new` were redefined in the corresponding Esirkepov code)
- Redefining `wqx` and `wqy` by doing a cos/sin combination of `ux`/`uy` (similar to what was done for `vy` in the corresponding Esirkepov routine).
Does that sound right?

Reply:
ok. will do